flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBResourceContainer.java

```diff
@@ -297,7 +298,9 @@ private DBOptions setDBOptionsFromConfigurableOptions(DBOptions currentOptions)
         currentOptions.setInfoLogLevel(internalGetOption(RocksDBConfigurableOptions.LOG_LEVEL));
 
         String logDir = internalGetOption(RocksDBConfigurableOptions.LOG_DIR);
-        if (logDir != null && !logDir.isEmpty()) {
+        if (logDir == null || logDir.isEmpty()) {
+            currentOptions.setDbLogDir(getFlinkLogDir());
```

Comment:
I think this method forgets to handle the case that `getFlinkLogDir` return `null`.